Minor exception cleanups.

MikaylaRay44 opened the pull request:
See #191, #193, #195, and #198.

Note: So I just realized some of these have already been handled and accepted in other pull requests, I can remove them if you'd like, but they are already accepted changes so let me know. (i.e. issue 191 was pull request 192).

mrapitis commented:
@MikaylaRay44 Can you please double check the for loops that have been replaced?

https://github.com/smartdevicelink/sdl_android/blob/feature/exception_cleanup/sdl_android_lib/src/com/smartdevicelink/util/BitConverter.java#L64 always initializes the hexArrayIndex value to 0.... its also generally preferred to initialize variables outside of the for loop.
https://github.com/smartdevicelink/sdl_android/blob/feature/exception_cleanup/sdl_android_lib/src/com/smartdevicelink/util/BitConverter.java#L84-85
https://github.com/smartdevicelink/sdl_android/blob/feature/exception_cleanup/sdl_android_lib/src/com/smartdevicelink/util/BitConverter.java#L64-67

We may also want to add some checking around offset values to avoid index out of bounds situations.

thanks

MikaylaRay44 commented:
Yes I don't know why I moved that index value, I will move it back.

As for the first for loop:

```java
for (int i = 0; i < hexString.length(); i += 2) {
	String byteString = hexString.substring(i, i + 2);
	byte theByte = (byte)Integer.parseInt(byteString, 16);
	theBytes[i/2] = theByte;
}
```

I didn't change this from how it appears on the develop branch, but I agree it looks concerning because of the downward cast.

I would suggest changing it to this to prevent potential loss of data, but it's not a *needed* change. I have added it to the pull request, I can remove it if you think it's not necessary.

```java
byte theByte = ((Integer) Integer.parseInt(byteString, 16)).byteValue();
```

Also the offset values do have checks in the Mime and BitConverter classes, but not all of them are documented so I've updated that. Did you see them in a different class?
